# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1289423095 18000
#      Wed Nov 10 16:04:55 2010 -0500
# Node ID 84846912f3c193c6ad2898d7926438b4feb4d496
# Parent  16d744cce38c7e59c1a9830227d79cba04f4238d
test/fntests.m: run tests for deprecated functions

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-11-10  John W. Eaton  <jwe@octave.org>
+
+	* scripts/deprecated/dispatch.m: Don't set warning state in tests.
+
 2010-11-10  John W. Eaton  <jwe@octave.org>
 
 	* deprecated/dispatch.m: Deprecate dispatch function.
 	* deprecated/module.mk (deprecated_FCN_FILES): Add it to the list.
 
 2010-11-09  John W. Eaton  <jwe@octave.org>
 
 	* help/help.m: Call missing_function_hook with output argument
diff --git a/scripts/deprecated/dispatch.m b/scripts/deprecated/dispatch.m
--- a/scripts/deprecated/dispatch.m
+++ b/scripts/deprecated/dispatch.m
@@ -45,61 +45,59 @@ function varargout = dispatch (varargin)
 
   varargout = cell (nargout, 1);
   [ varargout{:} ] = __dispatch__ (varargin{:});
 
 endfunction
 
 
 %!test # builtin function replacement
-%! warning off Octave:deprecated-function
 %! dispatch('sin','length','string')
 %! assert(sin("abc"),3)
 %! assert(sin(0),0,10*eps); 
+
 %!test # 'any' function
-%! warning off Octave:deprecated-function
 %! dispatch('sin','exp','any')
 %! assert(sin(0),1,eps);
 %! assert(sin("abc"),3);
+
 %!test # 'builtin' function
-%! warning off Octave:deprecated-function
 %! assert(builtin('sin',0),0,eps);
 %! builtin('eval','x=1;');
 %! assert(x,1);
+
 %!test # clear function mapping
-%! warning off Octave:deprecated-function
 %! dispatch('sin','string')
 %! dispatch('sin','any')
 %! assert(sin(0),0,10*eps);
+
 %!test # oct-file replacement
-%! warning off Octave:deprecated-function
 %! dispatch('fft','length','string')
 %! assert(fft([1,1]),[2,0]);
 %! assert(fft("abc"),3)
 %! dispatch('fft','string');
+
 %!test # m-file replacement
-%! warning off Octave:deprecated-function
 %! dispatch('hamming','length','string')
 %! assert(hamming(1),1)
 %! assert(hamming("abc"),3)
 %! dispatch('hamming','string')
 
 %!test # override preloaded builtin
-%! warning off Octave:deprecated-function
 %! evalin('base','cos(1);');
 %! dispatch('cos','length','string')
 %! evalin('base','assert(cos("abc"),3)');
 %! evalin('base','assert(cos(0),1,eps)');
 %! dispatch('cos','string')
+
 %!test # override pre-loaded oct-file
-%! warning off Octave:deprecated-function
 %! evalin('base','qr(1);');
 %! dispatch('qr','length','string')
 %! evalin('base','assert(qr("abc"),3)');
 %! evalin('base','assert(qr(1),1)');
 %! dispatch('qr','string');
+
 %!test # override pre-loaded m-file
-%! warning off Octave:deprecated-function
 %! evalin('base','hanning(1);');
 %! dispatch('hanning','length','string')
 %! evalin('base','assert(hanning("abc"),3)');
 %! evalin('base','assert(hanning(1),1)');
 %! dispatch('hanning','string');
diff --git a/test/ChangeLog b/test/ChangeLog
--- a/test/ChangeLog
+++ b/test/ChangeLog
@@ -1,8 +1,13 @@
+2010-11-10  John W. Eaton  <jwe@octave.org>
+
+	* fntests.m: Set warning state off for Octave:deprecated-functions.
+	(run_test_script): Do run tests for deprecated functions.
+
 2010-11-09  John W. Eaton  <jwe@octave.org>
 
 	* test_parser.m: New file.
 	* Makefile.am (FCN_FILES): Add it to the list.
 
 2010-09-29  Rik  <octave@nomad.inbox5.com>
 
 	* fntests.m: Break output text differently for better readability.
diff --git a/test/fntests.m b/test/fntests.m
--- a/test/fntests.m
+++ b/test/fntests.m
@@ -131,17 +131,17 @@ function [dp, dn, dxf, dsk] = run_test_s
   global files_with_no_tests;
   global topsrcdir;
   global topbuilddir;
   lst = dir (d);
   dp = dn = dxf = dsk = 0;
   for i = 1:length (lst)
     nm = lst(i).name;
     if (lst(i).isdir && ! strcmp (nm, ".") && ! strcmp (nm, "..")
-        && ! strcmp (nm, "CVS") && ! strcmp (nm, "deprecated") )
+        && ! strcmp (nm, "CVS"))
       [p, n, xf, sk] = run_test_script (fid, [d, "/", nm]);
       dp += p;
       dn += n;
       dxf += xf;
       dsk += sk;
     endif
   endfor
   for i = 1:length (lst)
@@ -210,16 +210,17 @@ function report_files_with_no_tests (wit
   printf ("\n%d (of %d) %s files have no tests.\n", n_without, n_tot, typ);
 endfunction
 
 pso = page_screen_output ();
 warn_state = warning ("query", "quiet");
 warning ("on", "quiet");
 try
   page_screen_output (0);
+  warning ("off", "Octave:deprecated-functions");
   fid = fopen ("fntests.log", "wt");
   if (fid < 0)
     error ("could not open fntests.log for writing");
   endif
   test ("", "explain", fid);
   dp = dn = dxf = dsk = 0;
   puts ("\nIntegrated test scripts:\n\n");
   for i = 1:length (fundirs)
